pkg/crc/network/nameservers.go

```diff
@@ -34,11 +36,35 @@ func GetResolvValuesFromInstance(sshRunner *ssh.Runner) (*ResolvFileValues, erro
 func UpdateResolvFileOnInstance(sshRunner *ssh.Runner, resolvFileValues ResolvFileValues) error {
 	resolvFile, _ := CreateResolvFile(resolvFileValues)
 
-	err := sshRunner.CopyDataPrivileged([]byte(resolvFile), "/etc/resolv.conf", 0644)
-	if err != nil {
-		return fmt.Errorf("Error creating /etc/resolv on instance: %s", err.Error())
+	sd := systemd.NewInstanceSystemdCommander(sshRunner)
+	// Check if ovs-configuration.service exist and if not then it is old bundle and use the same way to
+	// update resolve.conf file
+	if state, err := sd.Status("ovs-configuration.service"); err != nil || state == states.NotFound {
+		err := sshRunner.CopyDataPrivileged([]byte(resolvFile), "/etc/resolv.conf", 0644)
+		if err != nil {
+			return fmt.Errorf("Error creating /etc/resolv on instance: %s", err.Error())
+		}
+		return nil
 	}
-	return nil
+	if err := sd.Start("ovs-configuration.service"); err != nil {
+		return err
+	}
+
+	nameservers := resolvFileValues.NameServers[0].IPAddress
+	for _, n := range resolvFileValues.NameServers[1:] {
+		nameservers = fmt.Sprintf("%s,%s", nameservers, n.IPAddress)
+	}
```

Comment:
https://pkg.go.dev/strings#Join ?

Reply:
To use `strings.join` I have to create another slice to capture value `resolvFileValues.NameServer` otherwise I can't .directly use because of type mismatch,

```
strings.Join(resolvFileValues.NameServers,  "," )
```

Reply:
Hmm, the structure of the ResolvFileValues type might be a bit too complicated, maybe it could be changed to only contain strings instead of struct containing strings
Or you could add this API:
```
@@ -34,6 +35,7 @@ func GetResolvValuesFromInstance(sshRunner *ssh.Runner) (*ResolvFileValues, erro
 func CreateResolvFileOnInstance(sshRunner *ssh.Runner, resolvFileValues ResolvFileValues) error {
        resolvFile, _ := CreateResolvFile(resolvFileValues)
 
+       log.Infof("nameservers: %s", strings.Join(resolvFileValues.GetNameServers(), ","))
        err := sshRunner.CopyDataPrivileged([]byte(resolvFile), "/etc/resolv.conf", 0644)
        if err != nil {
                return fmt.Errorf("Error creating /etc/resolv on instance: %s", err.Error())
diff --git a/pkg/crc/network/types.go b/pkg/crc/network/types.go
index 0d7115a4b..dda58693f 100644
--- a/pkg/crc/network/types.go
+++ b/pkg/crc/network/types.go
@@ -20,6 +20,24 @@ type ResolvFileValues struct {
        NameServers   []NameServer
 }
 
+func (vals *ResolvFileValues) GetSearchDomains() []string {
+       searchDomains := []string{}
+       for _, domain := range vals.SearchDomains {
+               searchDomains = append(searchDomains, domain.Domain)
+       }
+
+       return searchDomains
+}
+
+func (vals *ResolvFileValues) GetNameServers() []string {
+       nameservers := []string{}
+       for _, ns := range vals.NameServers {
+               nameservers = append(nameservers, ns.IPAddress)
+       }
+
+       return nameservers
+}
+
 type Mode string
 
 func (m Mode) String() string {

```

If you want to keep the code you have, can you move it to a helper function? In my opinion, this is a very low-level implementation detail, which is not really interesting to have directly in CreateResolvFileOnInstance body. Your code will segfault if `NameServers` has 0 or 1 elements, this will also need to be fixed if you want to keep it.

Reply:
Updated with this api, should prevent segfault.